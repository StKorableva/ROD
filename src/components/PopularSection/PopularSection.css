.block_of_product_cards{
    background-color:  var(--main-black-color);
    color: #FFFFFF;
    letter-spacing: -0.3px;
    width: 100vw;
}

.block_of_product_cards a{
    color: #FFFFFF !important;
    text-decoration: none
    /*//TODO font-family: 'Avenir Next' походу платный шрифт 
    //https://www.fontshop.com/families/avenir-next
    */
}

.block_of_product_cards__heading_box{
    display: flex;
    align-items: center;
    justify-content: space-between;
}
.block_of_product_cards__heading_box-heading{
    font-family: 'Cormorant', serif;
    font-size: 32px;
    line-height: 39px;
    letter-spacing: -0.3px;
    font-style: italic;
    font-weight: lighter;
}
.block_of_product_cards__heading_box p{
    margin:10px 20px 10px 20px;
}

.block_of_product_cards__content_box{
    display: flex;
    justify-content: space-around;
}

.block_of_product_cards__heading_box-link{
    font-size: 12px;
    line-height: 16px;
    text-align: right;
    text-decoration-line: underline;
}