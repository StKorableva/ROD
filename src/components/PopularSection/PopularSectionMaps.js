function mapStateToProps(state) {
	return {
		fetching: state.fetching,
		heading: state.heading,
		data: state.data,
		//TODO нужен ли вообще тот стор?
	};
}

function mapDispatchToProps(dispatch) {
	return {
		addToFavoritesHandler: (id) =>
			dispatch({ type: 'ADD_TO_FAVORITES', id: id }),
		//TODO добавить необходимые методы
		//TODO нужен ли вообще тот стор?
	};
}

export { mapStateToProps, mapDispatchToProps };
