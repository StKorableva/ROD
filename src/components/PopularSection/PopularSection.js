import ProductCard from '../ProductCard/ProductCard';
import '../PopularSection/PopularSection.css';
//TODO передай в меня fetching, heading, data, addToFavoritesHandler, openProductPage и массив обьектов со свойствами liked, imgUrl, text, price, id

function PopularSection(props) {
	const { fetching, heading, data, addToFavoritesHandler, openProductPage } =
		props;

	const getDOM = function () {
		let newDOM = [];
		for (let i = 0; i < data.length; i++) {
			const { liked, imgUrl, text, price, id } = data[i];
			newDOM.push(
				<ProductCard
					liked={liked}
					openProductPage={() => {
						openProductPage();
						//TODO добавить метод для открытия страницы товара(мне пока не понятно в каком виде это у нас будет происходить)
					}}
					addToFavorites={() => {
						addToFavoritesHandler(id);
					}}
					imgUrl={imgUrl}
					text={text}
					price={price} //TODO как будет храниться прайс? его нужно как то преобразовывать в компоненте ?
					key={id}
				/>
			);
		}
		return newDOM;
	};

	if (fetching) {
		return <></>;
	} else {
		return (
			<div className="block_of_product_cards">
				<div className="block_of_product_cards__heading_box">
					<p className="block_of_product_cards__heading_box-heading">
						{heading}
					</p>
					<p className="block_of_product_cards__heading_box-link">
						<a href="./">Смотреть всё</a>
						{/* //TODO добавить адрес/функцию, мб передать пропсом извне  */}
					</p>
				</div>
				<div className="block_of_product_cards__content_box">{getDOM()}</div>
			</div>
		);
	}
}

export default PopularSection;
