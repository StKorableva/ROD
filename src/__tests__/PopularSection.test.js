import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import PopularSection from '../components/PopularSection/PopularSection';

const objectForTest = [
	{ liked: true, imgUrl: './', text: 'Test', price: 10000, id: 1 },
	{ liked: false, imgUrl: './', text: 'Test', price: 10000, id: 1 },
];

describe('base test', () => {
	const component = shallow(
		<PopularSection
			fetching={false}
			heading="Популярное"
			data={objectForTest}
		></PopularSection>
	);

	it('render test', () => {
		expect(component).toHaveLength(1);
	});

	it('snapshot test', () => {
		expect(toJSON(component)).toMatchSnapshot();
	});

	it(`contains product cards (${objectForTest.length})`, () => {
		expect(component.find('ProductCard')).toHaveLength(2);
	});
});

describe('/fetching/ reaction test', () => {
	const component = shallow(
		<PopularSection
			fetching={true}
			heading="Популярное"
			data={[]}
		></PopularSection>
	);
	const Fragment = <></>;
	it('return fragment when fetching=true', () => {
		expect(component.contains(Fragment));
	});
});
