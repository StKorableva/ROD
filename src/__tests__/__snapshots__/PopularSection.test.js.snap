// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base test snapshot test 1`] = `
<div
  className="block_of_product_cards"
>
  <div
    className="block_of_product_cards__heading_box"
  >
    <p
      className="block_of_product_cards__heading_box-heading"
    >
      Популярное
    </p>
    <p
      className="block_of_product_cards__heading_box-link"
    >
      <a
        href="./"
      >
        Смотреть всё
      </a>
    </p>
  </div>
  <div
    className="block_of_product_cards__content_box"
  >
    <ProductCard
      addToFavorites={[Function]}
      imgUrl="./"
      key="1"
      liked={true}
      openProductPage={[Function]}
      price={10000}
      text="Test"
    />
    <ProductCard
      addToFavorites={[Function]}
      imgUrl="./"
      key="1"
      liked={false}
      openProductPage={[Function]}
      price={10000}
      text="Test"
    />
  </div>
</div>
`;
